File: C++/509.斐波那契数.cpp
/*
 * @lc app=leetcode.cn id=509 lang=cpp
 *
 * [509] 斐波那契数
 */

// @lc code=start
class Solution
{
public:
    int fib(int n)
    {
        if (n < 2)
            return n;
        int p = 0, q = 0;
        for (int i = 2; i <= n; i++)
        {
            p = q;
            q = result;
            result = p + q;
        }
        return result;
    }

private:
    int result = 1;
};
// @lc code=end

File: JavaScript/509.斐波那契数.js
/*
 * @lc app=leetcode.cn id=509 lang=javascript
 *
 * [509] 斐波那契数
 */

// @lc code=start
/**
 * @param {number} n
 * @return {number}
 */
var fib = function (n) {
  const fibArr = new Array(n + 1).fill(null);
  fibArr[1] = 1;
  fibArr[2] = 1;
  for (let i = 3; i <= n; i++) {
    fibArr[i] = fibArr[i - 1] + fibArr[i - 2];
  }
  return fibArr[n];
};
// @lc code=end
